Documentation/ApiOverview/Environment/ApplicationContext.rst

```diff
@@ -0,0 +1,373 @@
+again and use it to set a different base URI using the
+:ref:`base variants <sitehandling-baseVariants>`:
+
+*   Development/Dev1 -> dev1.example.com
+*   Development/Dev2 -> dev2.example.com
+*   Development/Dev3 -> dev3.example.com
+*   Development/Dev4 -> dev4.example.com
+*   Development/Dev5 -> dev5.example.com
+
+..  _root-application-context:
+
+Root ApplicationContext
+=======================
+
+It doesn't matter whether you are working with just one `ApplicationContext`
+or with an `ApplicationContext` divided several times by a slash (`/`). The
+first part is always the root `ApplicationContext` and must always be either
+`Production`, `Development` or `Testing`, otherwise the `isProduction`,
+`isDevelopment` and `isTesting` methods will not work.
+
+..  _parent-application-context:
+
+Parent ApplicationContext
+=========================
+
+This section only applies if you have divided the `ApplicationContext` into
+several sections using slashes (`/`). The entire remaining value after the
+first slash is used to instantiate a new `ApplicationContext`. The so-called
+parent `ApplicationContext`. Here you can see how the context is nested:
+
+*   ApplicationContext: `Development/Local/Ddev/Dev2`
+    *   Root ApplicationContext: `Development`
+    *   Parent ApplicationContext: `Local/Ddev/Dev2`
+        *   Root ApplicationContext: `Local`
+        *   Parent ApplicationContext: `Ddev/Dev2`
+            *   Root ApplicationContext: `Ddev`
+            *   Parent ApplicationContext: `Dev2`
+
+As written above the root `ApplicationContext` must always be one of the 3
+values: `Production`, `Development` or `Testing`. With the 2nd nesting at the
+latest, the root `ApplicationContext` here is now `Local` and with the 3rd
+nesting `Ddev`. There is no way to query this `root ApplicationContext` in the
+PHP class :php:`ApplicationContext`! You only have the option of using
+`getParent()` to access the next parent `ApplicationContext` and using
+:php:`(string)getParent()` to return the complete `ApplicationContext` as a
+string. This means that `Local/Ddev/Dev2` is returned at level 2
+and `Ddev/Dev2` at level 3.
+
+..  _read-application-context:
+
+Reading the ApplicationContext
+==============================
+
+TYPO3 itself already queries the `ApplicationContext` in various places, but
+you can also react to the `ApplicationContext` in various places.
+
+..  _read-application-context-php:
+
+PHP
+---
+
+Here are a few examples of how to access the `ApplicationContext` with the
+:php:`Environment` class:
+
+..  code-block:: php
+
+    <?php
+    use TYPO3\CMS\Core\Core\Environment;
+    class SunnyProducts
+    {
+        public function getDiscount(): int
+        {
+            if (Environment::getContext()->isDevelopment()) {
+                return 20;
+            }
+
+            return 4;
+        }
+    }
+
+..  _read-application-context-site-configuration:
+
+Site Config
+-----------
+
```

Comment:
We should mention, it can only be used for base variants?

Reply:
I have added a description